docs/api.md

```diff
@@ -391,4 +392,90 @@ to read a new credential from this role, execute the following:
 
 ```shell
 vault read -tls-skip-verify github/raf-backstage-demo/token/one-repo-only
+```
+
+## VaultSecret
+
+The VaultSecret CRD allows a user to create a K8s Secret from one or more Vault Secrets. It uses go templating to allow formatting of the K8s Secret in the `output.stringData` section of the spec.
+
+> Note: if reading a dynamic secret you typically care to set the `refreshThreshold` only (not the `refreshPeriod`). For just Key/Value Vault secrets, set the `refreshPeriod`.
+> 
+> See https://www.vaultproject.io/docs/concepts/lease to understand lease durations.
+
+- `refreshPeriod` the pull interval for syncing Vault secrets with the K8s Secret. This settings takes precedence over any lease duration returned by vault, effectively controlling when exactly all vault secrets defined in the vaultSecretDefinitions should re-sync.
+- `refreshThreshold` this is will instruct the operator to refresh the K8s Secret when a percentage of the lease duration has elapsed, if no `refreshPeriod` is specified. This is particularly useful for controlling when dynamic secrets should be refreshed before the lease duration is exceeded. The default is 90, meaning the secret would refresh after 90% of the time has passed from the vault secret's lease duration. When multiple vaultSecretDefinitions are defined, the smallest lease duration will be used when performing the scheduling for the next refresh.
```

Comment:
why do you say: `if no refreshPeriod is specified`? this should always be applied regardless of how the refreshPeriod is calcutaled.

Reply:
refreshPeriod is an override and if set will always resync after the interval is passed

Reply:
agreed. I was referring to the comment on the refreshThreshold documentation. `refreshThreshold` should apply to `refreshPeriod` too.

Reply:
`refreshThreshold` is really only needed for dynamic vault secrets. I think it would be confusing to apply a threshold percentage to a `refreshPeriod`, especially when the default `refreshThreshold` is 90%.